lib/libuutil/uu_list.c

```diff
@@ -505,14 +505,20 @@ uu_list_walk(uu_list_t *lp, uu_walk_fn_t *func, void *private, uint32_t flags)
 	}
 
 	if (lp->ul_debug || robust) {
-		uu_list_walk_t my_walk;
+		uu_list_walk_t *my_walk;
 		void *e;
 
-		list_walk_init(&my_walk, lp, flags);
+		my_walk = calloc(1, sizeof (*my_walk));
```

Comment:
```suggestion
		my_walk = uu_zalloc(sizeof (*my_walk));
```

Would be consistent with the other allocations in `uu_list.c`.  Although it doesn't really matter.